components/brave_wallet_ui/common/async/hardware.ts

```diff
@@ -177,37 +177,54 @@ export async function signLedgerSolanaTransaction (
+    const result = await apiProxy.solanaTxManagerProxy.processSolanaHardwareSignature(txId, [...signedMessage])
     if (!result || !result.status) {
       return { success: false, error: getLocale('braveWalletProcessTransactionError') }
     }
 
     return { success: result.status }
 }
 
-export async function signMessageWithHardwareKeyring (vendor: HardwareVendor, path: string, messageData: BraveWallet.SignMessageRequest): Promise<SignHardwareMessageOperationResult> {
-  const deviceKeyring = getHardwareKeyring(vendor)
+export async function signMessageWithHardwareKeyring (vendor: HardwareVendor, path: string, messageData: BraveWallet.SignMessageRequest): Promise<SignHardwareTransactionOperationResult> {
+  const deviceKeyring = getHardwareKeyring(vendor, messageData.coin)
   if (deviceKeyring instanceof LedgerBridgeKeyring) {
     if (messageData.isEip712) {
       if (!messageData.domainHash || !messageData.primaryHash) {
         return { success: false, error: getLocale('braveWalletUnknownInternalError') }
       }
       return deviceKeyring.signEip712Message(path, messageData.domainHash, messageData?.primaryHash)
-    } else return deviceKeyring.signPersonalMessage(path, messageData.message)
+    }
+    return deviceKeyring.signPersonalMessage(path, messageData.message)
   } else if (deviceKeyring instanceof TrezorBridgeKeyring) {
     if (messageData.isEip712) {
       if (!messageData.domainHash || !messageData.primaryHash) {
         return { success: false, error: getLocale('braveWalletUnknownInternalError') }
       }
       return deviceKeyring.signEip712Message(path, messageData.domainHash, messageData.primaryHash)
-    } else return deviceKeyring.signPersonalMessage(path, messageData.message)
+    }
+    return deviceKeyring.signPersonalMessage(path, messageData.message)
+  } else if (deviceKeyring instanceof SolanaLedgerBridgeKeyring) {
+    // Not supported yet, see https://github.com/solana-labs/solana/issues/21366.
+    return { success: false, error: getLocale('braveWalletHardwareOperationUnsupportedError') }
   }
   return { success: false, error: getLocale('braveWalletUnknownKeyringError') }
 }
 
-export async function cancelHardwareOperation (vendor: HardwareVendor) {
-  const deviceKeyring = getHardwareKeyring(vendor)
-  if (deviceKeyring instanceof LedgerBridgeKeyring || deviceKeyring instanceof TrezorBridgeKeyring) {
+export async function signRawTransactionWithHardwareKeyring (vendor: HardwareVendor, path: string, message: BraveWallet.ByteArrayStringUnion, coin: BraveWallet.CoinType): Promise<SignHardwareTransactionOperationResult> {
+  const deviceKeyring = getHardwareKeyring(vendor, coin)
+
+  if (deviceKeyring instanceof SolanaLedgerBridgeKeyring && message.bytes) {
+    return deviceKeyring.signTransaction(path, Buffer.from(message.bytes))
+  } else if (deviceKeyring instanceof TrezorBridgeKeyring || deviceKeyring instanceof LedgerBridgeKeyring) {
```

Comment:
We should do the following renaming:
```
s/TrezorBridgeKeyring/EthereumTrezorBridgeKeyring/g
s/LedgerBridgeKeyring/EthereumLedgerBridgeKeyring/g
```
And shouldn't we also check `FilecoinLedgerKeyring` here? otherwise it would throw unknown keyring error

Reply:
I left the name of those keyrings unchanged as it's not in the scope of this PR and I would like to minimize the change for uplift.
Added filecoin keyring in https://github.com/brave/brave-core/pull/14099/commits/0e0b7170bbe033d942776003d4309f071fef19cd